src/extra/components/dld.py

```diff
@@ -269,32 +294,50 @@ def edges(self, channel_index=True, pulse_dim='pulseId'):
         else:
             return edges
 
-    def signals(self, pulse_dim='pulseId'):
+    def signals(self, pulse_dim='pulseId', extra_columns={}):
         """Get reconstructed signals as dataframe.
 
         Args:
             pulse_dim ({pulseId, pulseIndex, time}, optional): Label
                 for pulse dimension, pulse ID by default.
+            extra_columns (dict, optional): Mapping of column name to
+                pulse-indexed data to align and insert into dataframe,
+                must be re-indexable to internal pulse index. Data is
+                repeated for multiple entries of a single pulse.
```

Comment:
I think the columns have to be pandas Series, right? It might be worth saying this explicitly.

Series objects can also have a name attribute, so we could consider allowing a list and using `s.name` as the column name. But that can be done later.